Python Tutorial Exercises(Week 5)/#8.py

```diff
@@ -0,0 +1,85 @@
+#week 1.1
+n = 5
+print("\n".join([str(v * v) for v in range(n + 1) if v != 4]))
+
+#week 1.2
+s = "Tianchi!@&%#Ma"
+print("".join([i.upper() if i.islower() else i.lower() if i.isupper() else i for i in s]))
+
+#week 1.3
+a = "harry had a little lamb"
+b = "ha"
+count = 0
+c = len(b)
+print(len([i for i in range(len(a)) if a[i:i + c] == b]))
+
+#week 1.4
+store_products = ["fruit", "water", "bread", "toilet paper", "soap"]
+shopping_list = ["eggs", "tooth paste", "toilet paper", "ice cream", "water"]
+print([i for i in shopping_list if i in store_products][0])
+
+#week 1.5
+x = 8
+y = 30
+print([i for i in range(x,y) if i % 7 == 0][0])
+
+# week 2.1
+x = [["this", "is", "a", "nested"], ["list", "that"], ["will", "be", "printed", "out"]]
+print("\n".join([val for sublist in x for val in sublist]))
+
+# week2.2
+c = 6
+print("\n".join([i * "6" for i in range(1, c + 1)]))
+
+# week2.3
+b = 4
+
+print("\n".join([((b - i) * "* " + str(i) + ((i - 1) * f" {i}")) for i in range(b,0,-1)][::-1]))
+
+# week2.4
+y = [4, 8, 2, 12, 7, 10]
+odd_number = "List has an odd number"
+non_odd_number = "List does not have an odd number"
+
+print("\n".join([i if i == odd_number else non_odd_number for i in [i if i % 2 != 0 else odd_number for i in y]]))
```

Comment:
This isn't quite right. This should only print once for the list.

Reply:
Yeah, that is the problem we didn't solve last time. I'm still thinking about the solution for that. :/

Reply:
The last time is back to week 3 exercises.

Reply:
Ah that missed my radar. Can you fix that for Week 2 exercise 4 please, in this branch? You should be able to use your knowledge of `for/else/break` to solve for that.